### PR TITLE
Implement community board (user queries/mutations) [5/7]

### PR DESCRIPTION
**Pull request by isaachan100**
## Overview

Implement user queries and mutations for community board

## Changes Made

- added logic for community board user queries/mutations in UserRepo
- added queries:
  - readFlyer(...): User
  - followOrganization(...): User
  - unfollowOrganization(...): User
- added unit jest testing

## Test Coverage

- tested queries/mutations locally
- unit jest testing runs to completion

## Next Steps (delete if not applicable)

- implement categories for organizations and finish onboarding organizations

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -8,14 +8,15 @@ import { ApolloServer } from 'apollo-server-express';
...
+import FlyerRepo from './repos/FlyerRepo';
 import FlyerResolver from './resolvers/FlyerResolver';
 import MagazineRepo from './repos/MagazineRepo';
 import MagazineResolver from './resolvers/MagazineResolver';
 import NotificationRepo from './repos/NotificationRepo';
 import OrganizationResolver from './resolvers/OrganizationResolver';
 import PublicationResolver from './resolvers/PublicationResolver';
-import WeeklyDebriefRepo from './repos/WeeklyDebriefRepo';
 import UserResolver from './resolvers/UserResolver';
+import WeeklyDebriefRepo from './repos/WeeklyDebriefRepo';
```
Comment (snajima) [inline src/app.ts:19]: love that we appdev members know the alphabetical order

File: src/app.ts [hunk 2]
```diff
@@ -76,13 +77,15 @@ const main = async () => {
   }
 
   async function setupTrendingArticleRefreshCron() {
-    // Refresh trending articles once
+    // Refresh trending articles, magazines, and flyers once
     ArticleRepo.refreshTrendingArticles();
     MagazineRepo.refreshFeaturedMagazines();
-    // Refresh trending articles 12 hours
+    FlyerRepo.refreshTrendingFlyers();
+    // Refresh trending articles, magazines, and flyers eveyr 12 hours
```
Comment (snajima) [inline src/app.ts:84]: but we don't know how to spell ic

File: src/repos/UserRepo.ts [hunk 3]
```diff
@@ -65,6 +66,37 @@ const unfollowPublication = async (uuid: string, pubSlug: string): Promise<User>
...
+ * Adds organization slug to user's followedOrganizations.
+ * Requires: the user is not already following the organization.
+ */
+const followOrganization = async (uuid: string, slug: string): Promise<User> => {
+  const user = await UserModel.findOne({ uuid });
+
+  if (user) {
+    user.followedOrganizations.push(new OrganizationSlug(slug));
+    return user.save();
+  }
```
Comment (snajima) [inline src/repos/UserRepo.ts:79]: do we need this if statement?

File: src/repos/UserRepo.ts [hunk 4]
```diff
@@ -65,6 +66,37 @@ const unfollowPublication = async (uuid: string, pubSlug: string): Promise<User>
...
+  const user = await UserModel.findOne({ uuid });
+  if (user) {
+    const orgSlugs = user.followedOrganizations.map((orgSlugObj) => orgSlugObj.slug);
+    const orgIndex = orgSlugs.indexOf(orgSlug);
+
+    if (orgIndex === -1) return user;
+
+    user.followedOrganizations.splice(orgIndex, 1);
+    return user.save();
+  }
```
Comment (snajima) [inline src/repos/UserRepo.ts:96]: same here

File: src/repos/UserRepo.ts [hunk 5]
```diff
@@ -99,7 +141,26 @@ const appendReadArticle = async (uuid: string, articleID: string): Promise<User>
 };
 
 /**
- * Add a magazine to a user's readMagazines
+ * Add a magazine to a user's readFlyers
+ */
+const appendReadFlyer = async (uuid: string, flyerID: string): Promise<User> => {
+  const user = await UserModel.findOne({ uuid });
+
+  if (!user) return user;
```
Comment (snajima) [inline src/repos/UserRepo.ts:149]: if `(!user)` is true, then shouldn't the return type be `Promise<User | undefined>`? does this line not give u an error :o

Comment (snajima) [reply src/repos/UserRepo.ts:149]: what exactly does `user` contain when `(!user)` is true?

**Review by snajima — APPROVED** [review]
let's gooo i have a few questinos but that's about it

File: src/repos/UserRepo.ts (hunk 3 above)
Comment (isaachan100) [reply src/repos/UserRepo.ts:79]: i think it might return null otherwise

File: src/repos/UserRepo.ts (hunk 5 above)
Comment (isaachan100) [reply src/repos/UserRepo.ts:149]: i checked in graphql playground and it just returns null for every field queried

File: src/repos/UserRepo.ts (hunk 4 above)
Comment (isaachan100) [reply src/repos/UserRepo.ts:96]: should be checking if user is null or not

**Review by katesliang — DISMISSED** [review]
cool 💯

**Review by snajima — CHANGES_REQUESTED** [review]
ahh ok i understand the user null thing now but let's keep things consistent – in two places u check `if (user)` but in the third u check `if (!user)`

**Review by snajima — APPROVED** [review]
let's gooo prod

**Review by katesliang — APPROVED** [review]

**Pull request merged**